File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/models/DateInfo.java
package nerd.tuxmobil.fahrplan.congress.models;


public class DateInfo {

    public int dayIdx;

    public String date;

    public DateInfo(int dayIdx, String date) {
        this.dayIdx = dayIdx;
        this.date = date;
    }

    public int getDayIndex(String date) {
        return this.date.equals(date) ? dayIdx : -1;
    }

    @SuppressWarnings({"SimplifiableIfStatement", "EqualsReplaceableByObjectsCall"})
    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        DateInfo dateInfo = (DateInfo) o;
        if (dayIdx != dateInfo.dayIdx) {
            return false;
        }
        return date != null ? date.equals(dateInfo.date) : dateInfo.date == null;
    }

    @Override
    public int hashCode() {
        int result = dayIdx;
        result = 31 * result + (date != null ? date.hashCode() : 0);
        return result;
    }

    @Override
    public String toString() {
        return "dayIndex = " + dayIdx + ", date = " + date;
    }

}

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/models/DateInfo.kt
package nerd.tuxmobil.fahrplan.congress.models

import info.metadude.android.eventfahrplan.commons.temporal.Moment

/**
 * Represents a connection between given [date] and [dayIndex].
 * Useful to later ask for a [dayIndex] providing a date (see [getDayIndex]).
 */
data class DateInfo(

        val dayIndex: Int,
        val date: Moment

) {

    companion object {
        const val DAY_INDEX_NOT_FOUND = -1
    }

    /**
     * Retrieve day index of stored date.
     * @return if given [date] matches the stored [date]
     * returns stored [dayIndex], [DAY_INDEX_NOT_FOUND] otherwise.
     */
    fun getDayIndex(date: Moment) = if (this.date == date) dayIndex else DAY_INDEX_NOT_FOUND

}
